notebooks/supply_chain/interact_with_environment.ipynb

```diff
@@ -0,0 +1,777 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "*** Env status after tick 0 ***\n",
+      "  Storage summary:\n",
+      "    [Facility]      [remain] / [total]\n",
+      "    Supplier_001        9900 /   10000\n",
+      "    Supplier_002        9800 /   10000\n",
+      "    Warehouse_001      27000 /   30000\n",
+      "    Retailer_001       17880 /   20000\n",
+      "  Storage detail:\n",
+      "    [Facility]       [sku1] [sku2] [sku3]\n",
+      "    Supplier_001          0      0    100\n",
+      "    Supplier_002        100    100      0\n",
+      "    Warehouse_001      1000   1000   1000\n",
+      "    Retailer_001        700    719    701\n",
+      "  Manufacture status:\n",
+      "    [Facility]      \n",
+      "    Supplier_001    sku_id = 3, manufacturing = 0, unit cost = 1\n",
+      "    Supplier_002    sku_id = 1, manufacturing = 0, unit cost = 1\n",
+      "    Supplier_002    sku_id = 2, manufacturing = 0, unit cost = 0\n"
+     ]
+    }
+   ],
+   "source": [
+    "env.step(None)  # Tick 0. Activate the enviornment, do not send any action\n",
+    "show_status(env)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Send actions to the environment"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#################################################################################################\n",
+    "# Use variables to store frequently used ids. These static information can be found in the previous cells.\n",
+    "#\n",
+    "# Notations:\n",
+    "#   FID = id, UID = id\n",
+    "#   S = supplier, W = warehouse, R = retailer\n",
+    "#   MANU = manufacture, CONSUME = consumer, PROD = product\n",
+    "#   Number at the end of UID = SKU id\n",
+    "#################################################################################################\n",
+    "FID_S001 = 1            # Facility id of Supplier_001\n",
+    "FID_S002 = 8            # Facility id of Supplier_002\n",
+    "FID_W001 = 17           # Facility id of Warehouse_001\n",
+    "UID_S001_MANU3 = 7      # Manufacture unit of Supplier_001, SKU 3\n",
+    "UID_S002_MANU1 = 14     # Manufacture unit of Supplier_002, SKU 1\n",
+    "UID_S002_MANU3 = 16     # Manufacture unit of Supplier_002, SKU 3\n",
+    "UID_W001_CONSUME1 = 23  # Consumer unit of Supplier_002, SKU 1\n",
+    "UID_W001_PROD3 = 26     # Product unit of Warehouse_001, SKU 3\n",
+    "UID_W001_CONSUME3 = 27  # Consumer unit of Warehouse_001, SKU 3\n",
+    "UID_R001_CONSUME1 = 31  # Consumer unit of Warehouse_001, SKU 1\n",
+    "UID_R001_CONSUME3 = 34  # Consumer unit of Retailer_001, SKU 3"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Manufacture action\n",
```

Comment:
Introduce the meaning of the parameters of each kind of Action before the example?

Reply:
Good suggestion, I'll do that.